app/scripts/services/quota.js

```diff
@@ -0,0 +1,185 @@
+      });
+    };
+
+    var filterQuotasForResource = function(resource, quotas) {
+      if (resource.kind === 'Pod') {
+        return filterQuotasForPod(resource, quotas);
+      }
+      // We plan on having other resources that scopes will affect
+      return quotas;
+    };
+
+    var humanizeQuotaResource = $filter('humanizeQuotaResource');
+    var humanizeKind = $filter('humanizeKind');
+
+    var getResourceLimitWarnings = function(resource, quota){
+      var warnings = [];
+      var podTemplate = resource.kind === "Pod" ? resource : _.get(resource, 'spec.template');
+      if (!podTemplate) {
+        // Didn't have a pod template, so we don't care about this resource
+        return warnings;
+      }
+
+      // Otherwise this is a pod or something that creates pods so return warnings if we are at quota already
+      // for any of these:
+      _.each([
+        'cpu',
+        'memory',
+        'requests.cpu',
+        'requests.memory',
+        'limits.cpu',
+        'limits.memory'
+      ], function(type) {
+        var q = quota.status.total || quota.status;
+        if (!isNil(q.hard[type]) && q.hard[type] === q.used[type]) {
+          var details;
+          if (resource.kind === 'Pod') {
+            details = "You will not be able to create the " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "'.";
+          }
+          else {
+            details = "You can can still create " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "' but no pods will be created until resources are freed.";
+          }
+          warnings.push({
+            type: resource.kind === 'Pod' ? 'error' : 'warning',
+            message: 'You are at your quota for ' + humanizeQuotaResource(type).toLowerCase() + ' on pods.',
+            details: details,
+            links: [{
+              href: "/project/" + quota.metadata.namespace + "/quota",
+              label: "View Quota",
+              target: "_blank"
+            }]
+          });
+        }
+      });
+
+      return warnings;
+    };
+
+    var getQuotaWarnings = function(resources, quotas, clusterQuotas) {
+      var warnings = [];
+
+      if (!resources || !quotas) {
+        return warnings;
+      }
+
+      _.each(resources, function(resource) {
+        var filteredQuotas = filterQuotasForResource(resource, quotas);
+        var filteredClusterQuotas = filterQuotasForResource(resource, clusterQuotas);
+
+        var rgv = APIService.objectToResourceGroupVersion(resource);
+        var humanizedResource = APIService.kindToResource(resource.kind, true);
+        var humanizedKind = humanizeKind(resource.kind);
+        var quotaKey = "";
+        if (rgv.group) {
+          quotaKey = rgv.group + "/";
+        }
+        quotaKey += rgv.resource;
+        var warningsForQuota = function(quota) {
+          var q = quota.status.total || quota.status;
+          if(q.hard[quotaKey] !== undefined && q.hard[quotaKey] === q.used[quotaKey]) {
```

Comment:
`!isNil` here as well instead of `!== undefined`? For consistency with `getResourceLimitWarnings`.